tensorboard/webapp/metrics/effects/metrics_effects_test.ts

```diff
@@ -510,6 +546,43 @@ describe('metrics effects', () => {
 
         expect(fetchTimeSeriesSpy).toHaveBeenCalledTimes(2);
       });
+
+      it('does not send requests to experiments lacking a cards tag', () => {
+        store.overrideSelector(getActivePlugin, METRICS_PLUGIN_ID);
+        store.overrideSelector(selectors.getExperimentIdsFromRoute, [
+          'exp1',
+          'exp2',
+        ]);
+        store.overrideSelector(
+          selectors.getVisibleCardIdSet,
+          new Set(['card1', 'card2'])
+        );
+        provideCardFetchInfo([
+          {id: 'card1', tag: 'tagA'},
+          {id: 'card2', tag: 'tagB'},
+        ]);
+        store.refreshState();
+
+        const effectFetchTimeSeriesSpy = spyOn(
+          effects as any,
+          'fetchTimeSeries'
+        ).and.stub();
+
+        actions$.next(coreActions.manualReload());
+
+        expect(effectFetchTimeSeriesSpy).toHaveBeenCalledTimes(2);
+        expect(effectFetchTimeSeriesSpy).toHaveBeenCalledWith({
```

Comment:
It's extremely hard to reason why the test concludes that these should be the requests sent.  Some of the key test data (like in overrideTagMetadata and overrideRunToEid) are setup far from here. Maybe it would be helpful to leave a comment about all the requests that could have been made and identify why certain requests were filtered out.